src/lib/storage/reference_segment.hpp

```diff
@@ -28,8 +28,8 @@ class ReferenceSegment : public BaseSegment {
 
   size_t size() const final;
 
-  const std::shared_ptr<const PosList> pos_list() const;
-  const std::shared_ptr<const Table> referenced_table() const;
+  const std::shared_ptr<const PosList>& pos_list() const;
```

Comment:
Out of curiosity: does this have a performance impact? Can copying a shared ptr be expensive?

Reply:
You need to atomically increment a counter (and decrement it, when your copy goes out-of-scope). Thus, it is bloody expensive if you have more than one thread copying it.

Reply:
We are sure that there are no cases where the PosList is requested and the reference counter should be incremented?

Reply:
That's the caller's responsibility. If you want your own shared_ptr, create a copy.

Reply:
I have an offline question on that topic.